qcodes/dataset/descriptions/dependencies.py

```diff
@@ -257,6 +269,26 @@ def _to_dict(self) -> InterDependencies_Dict:
                                           "standalones": standalones}
         return output
 
+    def _empty_data_dict(self) -> Dict[str, Dict[str, np.ndarray]]:
+        """
+        Create an empty dictionary (with empty numpy arrays as values)
+        matching the expected output of get_parameter_data / cache.data
+        based on the interdeps in this class.
+        """
+
+        output: Dict[str, Dict[str, np.ndarray]] = {}
+        for key, value in self.dependencies.items():
+            ps_id = self._paramspec_to_id[key]
+            output[ps_id] = {ps_id: np.array([])}
+            dep_ids = [self._paramspec_to_id[ps] for ps in value]
+            for dep_id in dep_ids:
+                output[ps_id][dep_id] = np.array([])
+        standalones = [self._paramspec_to_id[ps] for ps in self.standalones]
+        for standalone in standalones:
```

Comment:
suggest 1) using paramspecs directly since ``.name`` does the job, while the notion of the "id" is less readable (for some historical reason that i already forgot), 2) more explicit variable names for readability of the logic
```suggestion
        for dependent, independents in self.dependencies.items():
            dependent_name = dependent.name
            output[dependent_name] = {dependent_name: np.array([])}
            for independent in independents:
                output[dependent_name][independent.name] = np.array([])
        for standalone in (ps.name for ps in self.standalones):
```